src/test/java/bio/terra/service/snapshotbuilder/query/QueryTest.java

```diff
@@ -128,42 +129,66 @@ void renderComplexSQL() {
   }
 
   @Test
-  void renderSQLWithDatasetModel() {
+  void renderSQLWithDatasetModelGCP() {
     DatasetModel dataset =
         new DatasetModel().name("name").dataProject("project").cloudPlatform(CloudPlatform.GCP);
-    TablePointer conceptTablePointer =
-        new TablePointer("concept", null, null, BigQueryVisitor.bqTableName(dataset));
+    Query query = buildGetConceptsQuery(100, BigQueryVisitor.bqTableName(dataset));
+    String sql = QueryTestUtils.collapseWhiteSpace(query.renderSQL());
+    String expected =
+        QueryTestUtils.collapseWhiteSpace(
+            """
+        SELECT c.concept_name, c.concept_id FROM `project.datarepo_name.concept` AS c
+        WHERE c.concept_id IN
+          (SELECT c.descendant_concept_id FROM `project.datarepo_name.concept_ancestor` AS c
+          WHERE c.ancestor_concept_id = 100)""");
+    assertThat(sql, is(expected));
+  }
+
+  @Test
+  void renderSQLWithDatasetModelAzure() {
+    String datasetSource = "source_dataset_data_source_0";
+    Query query = buildGetConceptsQuery(100, SynapseVisitor.azureTableName(datasetSource));
```

Comment:
Testing the code this way mixes two different tests together; one for query generation and one for `SynapseVisitor.azureTableName()`. To avoid testing both at once, I would instead pass in a custom table name generator that uses `Function.identity()` here.

Then add a test to `SynapseVisitorTest` (creating that class if necessary) that specifically tests the behavior of `SynapseVisitor.azureTableName()`.